docker_ci/Jenkinsfile

```diff
@@ -0,0 +1,73 @@
+/*
+This pipeline builds a Docker image of MOOSE, runs the standard suite of tests,
+and pushes the image to the registry prescribed by $DOCKER_TAG.  A second push 
+is also done, with the tag suffix set to the commit hash of the MOOSE working 
+tree.  For this all to work, $DOCKER_TAG must be prescribed as a pipeline 
+parameter in Jenkins.
+*/
+
+pipeline {
+    agent none
+    
+    stages {
+        
+        // Run 'docker build' on the Dockerfile in this repository
+        // with args for core count and submodule commit hashes
+        stage('Build Image')
+        {
+            agent any
+            
+            steps {
+                sh '''
+                # Setup for getting submodule commit hashes
+                git submodule init
+                sub_rev() { git submodule status $1 | awk '{print $1}' ; }
+                
+                # Build MOOSE image with max cores and compatible submodules
+                docker build \
+                --build-arg MOOSE_JOBS=$(grep -c ^processor /proc/cpuinfo) \
+                --build-arg PETSC_REV=$(sub_rev petsc) \
+                --build-arg LIBMESH_REV=$(sub_rev libmesh) \
+                -f "$WORKSPACE/docker_ci/Dockerfile" \
+                -t $DOCKER_TAG .
+                '''
+            }
+        }
+        
+        // Pipes output from MOOSE tests into a readable, accessible file
+        stage('Run Tests')
+        {
+            agent any
+            
+            steps {
+                sh '''
+                # Run tests and pipe to accessible log file
+                docker run $DOCKER_TAG /bin/bash -c \
+                'cd test; ./run_tests' | tee test_results.log
+                
+                # Remove characters from text coloring
+                sed -ie 's/\\x1b\\[[0-9;]*m//g' test_results.log
```

Comment:
How about using the `--no-color` option instead of striping characters after the fact?

Reply:
Ah, thanks.  Hadn't realized that was an option.  I'll update it.